### PR TITLE
Add `bbolt surgery revert-meta-page` command

### PR DESCRIPTION
**Pull request by ahrtr**
Linked to https://github.com/etcd-io/bbolt/issues/370

Note that the first commit is based on https://github.com/etcd-io/bbolt/pull/384. Once 384 is merged, I will rebase this PR.

### CODE REVIEW

**Review by ptabor — APPROVED** [review]
Thank you.

File: cmd/bbolt/surgery_commands.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+	// Ensure output file not exist.
+	_, err = os.Stat(cmd.DstPath)
+	if err == nil {
+		return fmt.Errorf("output file %q already exists", cmd.DstPath)
+	} else if !os.IsNotExist(err) {
+		return err
+	}
+
+	// Copy database from SrcPath to DstPath
```
[the remark this answers is not in the document]
Comment (ahrtr) [reply cmd/bbolt/surgery_commands.go:117]: Good idea. Done.

File: cmd/bbolt/surgery_commands_test.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	dstPath := filepath.Join(t.TempDir(), "dstdb")
+	m := NewMain()
+	err = m.Run("surgery", "revert-meta-page", "-o", dstPath, srcPath)
+	require.NoError(t, err)
+
+	// read both meta0 and meta1 from dst file
+	dstBuf0, dstBuf1 := readBothMetaPages(t, dstPath, pageSize)
+
+	// check result. Note we should skip the page ID
+	assert.Equal(t, nonActiveSrcBuf[8:], dstBuf0[8:])
```
[the remark this answers is not in the document]
Comment (ahrtr) [reply cmd/bbolt/surgery_commands_test.go:48]: Updated.

File: cmd/bbolt/surgery_commands.go [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    help                   print this screen
+    revert-meta-page       revert the meta page change made by the last transaction
+
+Use "bbolt surgery [command] -h" for more information about a command.
+`, "\n")
+}
+
+// RevertMetaPageCommand represents the "surgery revert-meta-page" command execution.
+type RevertMetaPageCommand struct {
+	baseCommand
```
[the remark this answers is not in the document]
Comment (ahrtr) [reply cmd/bbolt/surgery_commands.go:66]: Let me do this when I add the next surgery command.

File: cmd/bbolt/surgery_commands_test.go [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"go.etcd.io/bbolt/internal/btesting"
+	"go.etcd.io/bbolt/internal/guts_cli"
+)
+
+func TestSurgery_RevertMetaPage(t *testing.T) {
+	pageSize := os.Getpagesize()
+	db := btesting.MustCreateDB(t)
```
[the remark this answers is not in the document]
Comment (ahrtr) [reply cmd/bbolt/surgery_commands_test.go:17]: We can add more test in a separate PR.

**Comment by ahrtr** [comment]
Just rebased this PR and address review comments. Let's consider to add more unit test later in separate PR.

**Review by ptabor — APPROVED** [review]

File: cmd/bbolt/surgery_commands_test.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (ahrtr) [reply cmd/bbolt/surgery_commands_test.go]: Updated it. thx

**Pull request merged**